prov/util/src/util_mr_map.c

```diff
@@ -219,10 +219,39 @@ static struct fi_ops ofi_mr_fi_ops = {
 	.ops_open = fi_no_ops_open
 };
 
+void ofi_mr_update_attr(uint32_t user_version, uint64_t caps,
+			const struct fi_mr_attr *user_attr,
+			struct fi_mr_attr *cur_abi_attr)
+{
+	cur_abi_attr->mr_iov = (struct iovec *) user_attr->mr_iov;
+	cur_abi_attr->iov_count = user_attr->iov_count;
+	cur_abi_attr->access = user_attr->access;
+	cur_abi_attr->offset = user_attr->offset;
+	cur_abi_attr->requested_key = user_attr->requested_key;
+	cur_abi_attr->context = user_attr->context;
+
+	if (FI_VERSION_GE(user_version, FI_VERSION(1, 5))) {
+		cur_abi_attr->auth_key_size = user_attr->auth_key_size;
+		cur_abi_attr->auth_key = user_attr->auth_key;
+	} else {
+		cur_abi_attr->auth_key_size = 0;
+		cur_abi_attr->auth_key = NULL;
+	}
+
+	if (caps & FI_HMEM) {
```

Comment:
FI_HMEM is not listed as a domain capability. May have to change to ```FI_VERSION_GE(user_version, FI_VERSION(1, 10))```.

Reply:
The version doesn't work, based on the man pages.  Relying on just the version ends up being difficult on developers.  If there's a feature that 1.10 offers that they want, we want to make it easy for them to use it without also forcing code changes in places that they don't care about.  For example, if they update to 1.10 to get collective support, they shouldn't need to change how they register memory as a result.

FI_HMEM support isn't a domain capability, but it's an fi_info cap.  That's passed in when opening the domain, which gets captured in the util 'info_domain_caps'.